ci/kind/kind-setup.sh

```diff
@@ -244,25 +245,53 @@ function configure_extra_networks {
   done
 }
 
+# add_kind_ipam_routes  add routes for non-ipam test-pods.
+function add_kind_ipam_routes {
+  kubectl get nodes -o jsonpath='{range .items[*]}{.spec.podCIDR}{" "}{.status.addresses[?(@.type=="InternalIP")].address}{"\n"}{end}' |\
+  while read pod_cidr node_ip; do
+    docker_run_with_host_net ip route add "$pod_cidr" via "$node_ip"
+  done
+}
+
+# delete_kind_ipam_routes delete the routes of non-ipam test-pods from the Kind testbed.
+function delete_kind_ipam_routes { 
+  cluster=$(kind get clusters)
```

Comment:
Can we only delete routes for the target cluster? I doubt this will work as expected if there are multiple Kind clusters

Reply:
You didn't get my point, no matter you check `FLEXIBLE_IPAM` or not, the `kind get clusters` will return an array if there are multiple clusters, how do you guarantee that your current context is the one you want to delete when you run `kubectl get nodes`?

Reply:
In kind-ipam has some special configurations and its different from other kind testbeds, in every ipam job build workflow will be 

- destroy previous ipam kind cluster (if exist dangling cluster )
- create new ipam kind cluster
- run test
- destroy current cluster

multiple cluster will only be possible  when someone abort the jobs and the my purpose of using this cmd `kind get clusters`  to check if there is any kind cluster (doesn't matter it return array or single cluster) if any cluster is there then just use "kubectl get nodes......" cmd to get cidr of that cluster ( because I observed that every kind cluster has same cidr subnet) and delete the routes if present on testbed.

why I am checking for cluster because for every build  we run Destroy function first and if there is no cluster available on testbed then you cannot run directly `kubectl get nodes.......` cmd to get cidrs it will show error , so to delete the route you need cidr to check whether route is present or not if present then delete otherwise no need to do so.


and suppose someone aborted the jobs and cluster is still there , then you trigger again ipam job then it will first destroy existing route and cluster.... then create new one, still I want just cidr to delete the route added by previous job if its present there and every cluster has same cidr.


for now we cannot enable Multiple jobs run because in the testbed we are using custom kind bridge network and its common bridge so if we allow multiple run then it will create issue because sub-interfaces for kind-1 already exist so we can create new one with same subnet and same vlan-id using common bridge.

not sure if I explain it well or not, but my purpose to use this cmd `kind get clusters ` to check any cluster is there or not. if cluster there then I can get cidrs using kubectl cmd and I can check for routes present or not, if there are any other way to  do it you can suggest me I will optimize it?

Reply:
@luolanzone 
It is taking care of deleting dangling ipam routes (if any) during destroying previous ipam kind cluster before creating a new one, 
cluster=$(kind get clusters) returns array of clusters, in the current implementation, deletion of routes are not specific to cluster/cluster context. Currently ipam kind test bed supports only single job run.
@KMAnju-2021 can create an issue to handle it in another PR since #5734 is closed.

Reply:
@tnqn one small doubt I think with the current kind-setup.sh script we cannot enable multiple kind job run (parallel builds) for those tests that are creating sub-interfaces using common kind bridge network using same subnet and vlan id could you correct me?

Reply:
> @tnqn one small doubt I think with the current kind-setup.sh script we cannot enable multiple kind job run (parallel builds) for those tests that are creating sub-interfaces using common kind bridge network using same subnet and vlan id could you correct me?

I guess so, that's one reason why I suggested not to pursude running multiple kinds in a single VM, and suggested to explore one kind per VM, or kind in docker.

Reply:
I do not think multiple kind jobs can run with a common kind docker network on a single vm, Ideally each job should have their own kind network. It was part of initial discussion. There isn't any issue creating multiple kind docker networks with different name on same test bed. #6468 (WIP) will address this issue. IPAM related issues will be addressed by #6545